src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -305,8 +295,13 @@ public class XmlConvert
         }
 
         private const int EncodedCharLength = 7; // ("_xFFFF_".Length);
-        private static volatile Regex? s_encodeCharPattern;
-        private static volatile Regex? s_decodeCharPattern;
+
+        [RegexGenerator("_[Xx][0-9a-fA-F]{4}(?:_|[0-9a-fA-F]{4}_)")]
+        private static partial Regex DecodeCharRegex();
+
+        [RegexGenerator("(?<=_)[Xx][0-9a-fA-F]{4}(?:_|[0-9a-fA-F]{4}_)")]
```

Comment:
@joperezr, note this pattern has a lookbehind.

@krwq, I left all of the patterns the same, except for this one, which I tweaked to be more efficient by factoring out the 4 repeater from the alternation and distributing the following underscore into the alternation.  I also changed the capture group to be non-capturing because the captures aren't actually used.

Reply:
It's the same (other than the lookbehind at the beginning).

Reply:
>@joperezr, note this pattern has a lookbehind.

Yeah, I guess that means we need your other change in first in order for this to work right? I imagine the generated code for this right now is simply just calling the Regex ctor until your change goes in?

Reply:
> Is this a transformation we would do internally? (the factoring)

There's an open issue for it.
https://github.com/dotnet/runtime/issues/64097#issuecomment-1019397246